lib/routes/account.js

```diff
@@ -985,33 +1041,61 @@ module.exports = function (
         log.begin('Account.RecoveryEmailResend', request)
         var sessionToken = request.auth.credentials
         var service = request.payload.service || request.query.service
-        if (sessionToken.emailVerified ||
-            Date.now() - sessionToken.verifierSetAt < config.smtp.resendBlackoutPeriod) {
-          return reply({})
-        }
-        customs.check(
-          request.app.clientAddress,
-          sessionToken.email,
-          'recoveryEmailResendCode')
-          .then(
-            mailer.sendVerifyCode.bind(
-              mailer,
-              sessionToken,
-              sessionToken.emailCode,
-              {
-                service: service,
-                redirectTo: request.payload.redirectTo,
-                resume: request.payload.resume,
-                acceptLanguage: request.app.acceptLanguage
+        var mailerFunc = null
+
+        return db.sessionTokenWithVerificationStatus(sessionToken.id)
+          .then(function (token) {
+            if (sessionToken.emailVerified) {
+              // Send sign-in verification email with code as sessionId
+              if (Date.now() - sessionToken.verifierSetAt < config.smtp.resendBlackoutPeriod) {
```

Comment:
I wonder if we need a copy of this `resendBlackoutPeriod` check on the `sendVerifyLoginEmail` branch as well, perhaps keyed off the createAt timestamp of the token.